src/js/components/modals/AddRepositoryFormModal.js

```diff
@@ -81,6 +90,20 @@ class AddRepositoryFormModal extends mixin(StoreMixin) {
         writeType: 'input',
         validation: /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)$/,
         value: ''
+      },
+      {
+        fieldType: 'number',
+        name: 'priority',
+        placeholder: 'Priority',
+        required: false,
+        min: '0',
+        max: `${numberOfRepositories}`,
+        step: '1',
+        validationErrorText: `Must be a positive integer between 0 and ${numberOfRepositories} representing its priority. 0 is the highest and ${numberOfRepositories} denotes the lowest priority.`,
```

Comment:
Nope, it will throw a range error if you go higher than that, so I figured it was a better experience to catch it up front